### PR TITLE
Add modular scss

### PR DESCRIPTION
**Pull request by salielim**

### CODE REVIEW

File: app/assets/stylesheets/components/_sidebar.scss [hunk 1]
```diff
@@ -0,0 +1,74 @@
+/* sidebar */
```
Comment (roberthopman) [inline app/assets/stylesheets/components/_sidebar.scss]: Can be removed.

File: app/assets/stylesheets/custom.css.scss [hunk 2]
```diff
@@ -1,314 +1,14 @@
...
+@import "base/universal";
+@import "base/typography";
+@import "base/debug";
+
+@import "components/header";
+@import "components/footer";
+@import "components/sidebar";
+@import "components/forms";
+@import "components/microposts";
+@import "components/users_index";
```
Comment (roberthopman) [inline app/assets/stylesheets/custom.css.scss]: What is the right order to load the base & components? cc @emile

**Review by roberthopman — COMMENTED** [review]
nice refactoring, 2 comments left 👍

File: app/assets/stylesheets/custom.css.scss (hunk 2 above)
Comment (salielim) [reply app/assets/stylesheets/custom.css.scss]: i wonder about this too

File: app/assets/stylesheets/base/_fonts.scss [hunk 3]
```diff
@@ -0,0 +1,2 @@
+$gray-medium-light: #eaeaea;
+$gray-light: #D3D3D3;
```
Comment (emilebosch) [inline app/assets/stylesheets/base/_fonts.scss]: put in `_vars`

File: app/assets/stylesheets/base/_typography.scss [hunk 4]
```diff
@@ -0,0 +1,26 @@
+@import "fonts";
```
Comment (emilebosch) [inline app/assets/stylesheets/base/_typography.scss]: no need to import

File: app/assets/stylesheets/components/_users_index.scss [hunk 5]
```diff
@@ -0,0 +1,9 @@
+.users {
```
Comment (emilebosch) [inline app/assets/stylesheets/components/_users_index.scss]: modules/_user-list.scss

.mod-user-list {

}

File: app/assets/stylesheets/custom.css.scss (hunk 2 above)
Comment (emilebosch) [reply app/assets/stylesheets/custom.css.scss]:
```
@import "base/mixins";
 +@import "base/universal";
 +@import "base/typography";
 +@import "base/debug";
 +
 +@import "components/*";
 +@import "modules/*";

```

**Review by emilebosch — CHANGES_REQUESTED** [review]
Fix this:
- modules
- components
- vars
- remove excessive imports

File: app/assets/stylesheets/components/_footer.scss [hunk 6]
```diff
@@ -0,0 +1,25 @@
+@import "base/fonts";
```
Comment (roberthopman) [inline app/assets/stylesheets/components/_footer.scss]: do we need to import this, if it's already loaded? (base is before components) @emilebosch ?

**Review by roberthopman — COMMENTED** [review]
just 1 comment

File: app/assets/stylesheets/components/_footer.scss (hunk 6 above)
Comment (emilebosch) [reply app/assets/stylesheets/components/_footer.scss]: try to remove it and see what it does!

SPOILER: If its loaded no need to `@import` it again

**Pull request closed without merging**